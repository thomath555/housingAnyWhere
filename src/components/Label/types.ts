import { FontSizes, FontWeights } from '../../utils/types/enums';

export type Props = {
  label: string;
  fontSize?: FontSizes;
  fontWeight?: FontWeights;
  color?: string;
};
