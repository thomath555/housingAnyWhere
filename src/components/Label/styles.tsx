/* eslint-disable @typescript-eslint/indent */
import styled, { css } from 'styled-components';

// Other
import colors from '../../assets/styles/colors';
import { FontSizes, FontWeights } from '../../utils/types/enums';

// Types
import { Props } from './types'

export const LabelWrapper = styled.label<Props>`
  display: block;
  position: relative;
${({ fontWeight = FontWeights.default }) =>
    css`
  font-weight: ${fontWeight};
`}

${({ color = '' }) =>
    css`
  color: ${color || colors.black};
`}

${({ fontSize = FontSizes.default }) =>
    css`
    font-size: ${fontSize}rem;
  `}
  `
