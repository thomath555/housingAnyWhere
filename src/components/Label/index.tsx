// Core
import React from 'react';

// Styles and Assets
import { LabelWrapper } from './styles';

// Types
import { Props } from './types';

const Label: React.FC<Props> = (props: Props) => {
  const {
    label
  } = props;
  return (
    <LabelWrapper {...props}>
      {label}
    </LabelWrapper>
  );
};

export default Label;
