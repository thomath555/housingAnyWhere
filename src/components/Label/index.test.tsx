import React from 'react'
import renderer from 'react-test-renderer'

import Label from './index'

describe('Component: Label', () => {
  // Snapshot test
  it('should render', () => {
    const component = <Label label='component Label testing' />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})