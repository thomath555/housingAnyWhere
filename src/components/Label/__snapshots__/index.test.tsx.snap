// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Label should render 1`] = `
<label
  className="sc-bdnxRM idbduD"
  label="component Label testing"
>
  component Label testing
</label>
`;
